src/VisualStudio/Core/Def/Implementation/UnusedReferences/ProjectAssets/ProjectAssetsReader.cs

```diff
@@ -121,6 +129,15 @@ internal static class ProjectAssetsReader
                     _ => ReferenceType.Assembly
                 };
 
+                if (referenceType == ReferenceType.Project &&
+                    library.Path is not null)
+                {
+                    // Project references are keyed by their filename but the
+                    // item specification should be the path to the project file
+                    // with Windows-style directory separators.
+                    itemSpecification = library.Path.Replace('/', '\\');
```

Comment:
Instead of constructing the ReferenceInfo with an ItemSpec of the files name (`dotnet-format`) we will construct it with the path to the project file (`..\\src\\dotnet-format.csproj`)